modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -160,17 +171,29 @@ func isAgent(bd *fleet.BundleDeployment) bool {
 	return strings.HasPrefix(bd.Name, "fleet-agent")
 }
 
-func shouldRedeploy(bd *fleet.BundleDeployment) bool {
+func shouldRedeploy(bd *fleet.BundleDeployment, status *fleet.BundleDeploymentStatus) (ok bool, duration time.Duration, err error) {
```

Comment:
Same. You can make it only return `bool` to improve readability. `ApplyInterval` and err(error can be avoided by changing the api) can be computed/avoided.

Reply:
done